### DISCUSSION TITLE
Understanding issue (solidity interface) !!

### DISCUSSION THREAD

**Discussion by abdullahansari-yk** [Cyfrin/foundry-full-course-cu#2542] [Q&A]
Guys, i am not getting solidity interface right now .
should i move on to next part and after completion,return to it ??
or
should i stick to it until i fully grasp is ??

HELP !!

**Comment by EngrPips** [comment]
Hello @ab6955, The Solidity interface is just a file used to ensure that you correctly implement every function you want a contract to have. Imagine you want to build a house, and you have a list of things you want to ensure the house has, so you write all those things on paper and then give every worker, constructor, and engineer working on your land to build the house. That paper, in which you write everything you want the house to have and give to the people working to build the house, is the interface that will guide all the workers to ensure that they do not forget any details and build the house to have everything you planned it to have. So when you do the below, you are simply saying that the `BigBoy` contract must implement every function inside the `ISwag` interface.

```js
contract BigBoy is ISwag {
}
```

Please ask questions if anything is confusing.

**Comment by mumernisar** [comment]
Functions mentioned within a interface are all external.. external functions are those that can only be called from contracts other than the one that implements the Interface...!
If a conteact is implementing an interface then it must provide implementation for all the functions within that contract..!!

See an example 
```solidity
interface MyInterface {
    function foo() external;
    function bar(uint256 _value) external returns (uint256);
}

// Implement the interface in a contract
contract MyContract is MyInterface {
    function foo() external override {
        // Implementation of foo
    }

    function bar(uint256 _value) external override returns (uint256) {
        // Implementation of bar
        return _value * 2;
    }
}
```
